### PR TITLE
Add useSmartClickHandler

### PR DESCRIPTION
**Pull request by peterhass**
https://github.com/good-hood-gmbh/nebenan-frontend/pull/1149#discussion_r475557748

### CODE REVIEW

File: src/use_anchor_sensitive_click_handler/index.es [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    const { target } = event;
+    const href = target.getAttribute('href');
+    const isLink = target.tagName === 'A';
+
+    if (isLink && href) {
+      event.preventDefault();
+      const isSameDomain = href.startsWith(global.location.origin);
+      const path = stripOriginFromUrl(href, global.location.origin);
+      if (isSameDomain) history.push(path);
+      else global.open(href);
```
Comment (peterhass) [inline src/use_anchor_sensitive_click_handler/index.es]: We should parse the anchor some more to respect stuff like `noopener`

Comment (max-degterev) [reply src/use_anchor_sensitive_click_handler/index.es]: -> `useSmartClickHandler`?

we run that on generated content, like markdown or smartcontent, we don't need to care about attributes

**Review by max-degterev — APPROVED** [review]
See 1 comment

**Review by max-degterev — APPROVED** [review]
LGTM

**Pull request merged**